### PR TITLE
Feature: Import and save XDITA files in Prosemirror-Jdita

### PR DESCRIPTION
**Pull request by plutonik-a**
## Purpose of this PR

The goal was to perform a round trip of uploading an XDita document and provide a download option of the current file as a JDita document.

The current process of transformation is `Xdita` -> `Jdita` -> `Prosemirror document` -> display in editor -> 
`Prosemirror document` -> `Jdita`

In a next feature we aim to complete the round trip of transforming the JDita document back to a complete XDita document.

The Prosemirror editor has been extended with a download button, so that users are able to download the current file displayed in the editor.

Screenshot 2024-05-08 at 17 26 47

### 1. Automated testing

Run the Cypress UI test:

```
yarn run test:demo
```

### 2. Manual UI test

Start the local server 

```
yarn run start:demo
```

Then open `http://localhost:1234`, or the port that is displayed in the console.

Click on the button in the menu "Open File" and browse in any XDita file, or choose one of our example files, that can be found here: `prosemirror-jdita-demo/example-xdita`

Now to start the download, click on button "Download File" and check the content of the downloaded JDita file.

---

**Note**   
This PR has been based on https://github.com/evolvedbinary/prosemirror-jdita/pull/233 and needs to be rebased accordingly.

### CODE REVIEW

File: prosemirror-jdita/src/tests/document.spec.ts [hunk 1]
```diff
@@ -102,7 +102,216 @@ describe('Function unTravel()', () => {
...
+      const transformedJdita = document(originalJdita);
+      // untransform the transformed jdita document
+      const result = unTravel(transformedJdita);
+
+      //compare the original jdita with the result
+      expect(result).to.deep.equal(originalJdita);
+    });
+
+    it.skip('handles complex jdita document', async () => {
+      const inputXml = `<?xml version="1.0" encoding="UTF-8"?>
```
Comment (plutonik-a) [inline prosemirror-jdita/src/tests/document.spec.ts:134]: @marmoure Could you please replace the content of `inputXml` with the constant `XML` in test-utils.ts?  See https://github.com/evolvedbinary/prosemirror-jdita/blob/feature/import-and-save-file/prosemirror-jdita/src/tests/test-utils.ts#L27-L90 
The code will be easier to read.

Comment (marmoure) [reply prosemirror-jdita/src/tests/document.spec.ts:134]: @plutonik-a I used multiple examples, I don't think i can rely on the constant `XML` should we be making more constants?

Comment (plutonik-a) [reply prosemirror-jdita/src/tests/document.spec.ts:134]: @marmoure Sure, for the reading convenience, you could move all xml example code into test-utils.ts. 
The small examples don't hurt inside the test file from my point of view, but I thought the huge document is just too much.

**Comment by marmoure** [comment]
Moved TO #283

**Pull request closed without merging**